riscv/insns/sret.h

```diff
@@ -13,7 +13,6 @@ reg_t prev_prv = get_field(s, MSTATUS_SPP);
 s = set_field(s, MSTATUS_SIE, get_field(s, MSTATUS_SPIE));
 s = set_field(s, MSTATUS_SPIE, 1);
 s = set_field(s, MSTATUS_SPP, PRV_U);
-STATE.sstatus->write(s);
```

Comment:
Is there any reason to move the `sstatus` write further down in the code?

I can't see any problem that doing so might cause, but I've seen subtle bugs result from reordering things like this. Absent any reason, I'd say the `sstatus` write should stay here.

Reply:
There's a data dependency on `prev_virt`, which is needed by Zicfilp to determine the `xLPE` state prior to the trap, so I have to place the update to `sstatus` later than all updates to `prev_virt`, which includes that read from `hstatus` in HS-mode. Though that being the reason, I know your concern, and that's why I did not move down other accesses to `sstatus` to keep them together, like this:

```c++
bool prev_virt = STATE.v;
if (!STATE.v) {
  if (p->extension_enabled('H')) {
    prev_virt = get_field(prev_hstatus, HSTATUS_SPV);
    reg_t new_hstatus = set_field(prev_hstatus, HSTATUS_SPV, 0);
    STATE.hstatus->write(new_hstatus);
  }

  STATE.mstatus->write(set_field(STATE.mstatus->read(), MSTATUS_MPRV, 0));
}
reg_t s = STATE.sstatus->read();
reg_t prev_prv = get_field(s, MSTATUS_SPP);
s = set_field(s, MSTATUS_SIE, get_field(s, MSTATUS_SPIE));
s = set_field(s, MSTATUS_SPIE, 1);
s = set_field(s, MSTATUS_SPP, PRV_U);
if (p->extension_enabled(EXT_ZICFILP)) {
  if (ZICFILP_READ_XLPE(prev_virt, prev_prv)) {
    STATE.elp = static_cast<elp_t>(get_field(s, SSTATUS_SPELP));
  } else {
    STATE.elp = elp_t::NO_LP_EXPECTED;
  }
  s = set_field(s, SSTATUS_SPELP, elp_t::NO_LP_EXPECTED);
}
STATE.sstatus->write(s);
```

However, since you took this issue up, perhaps I can ask if you would like me to change the code to something like the above in this case? Maybe for better readability?

Reply:
No, it's good like it is. Thank you for the explanation.